doc/source/ray-core/scheduling/ray-oom-prevention.rst

```diff
@@ -16,117 +16,80 @@ What is the memory monitor?
-  It will retry indefinitely if this value is set to -1.
+Using the Memory Monitor
+------------------------
 
-  When the task is retried due to OOM it applies a delay before re-executing the task. The delay is calculated as
+Retry policy
+~~~~~~~~~~~~
 
-  .. code-block:: bash
+When a task or actor is killed by the memory monitor it will be retried with exponential backoff. The retry delay is calculated as
 
-      delay_seconds = 2 ^ attempt
+.. code-block:: bash
 
-  Where the first retry will be delayed by 1 second as ``attempt`` starts from 0.
+    delay_seconds = 2 ^ (retry_attempt - 1)
 
-  .. note::
+The first ``retry_attempt`` will be delayed by 1 second. There is a cap on the maximum delay, which is 60 seconds.
 
-      Keep the value of ``RAY_task_oom_retries`` low, below 25, to avoid extremely long delays as it is using exponential backoff.
-      The delay is capped at 1 minute.
-  
+This retry is applied only when the worker is killed by the memory monitor. Otherwise, when the worker dies, it uses :ref:`max_retries <task-fault-tolerance>`, or :ref:`max_restarts <actor-fault-tolerance>` and :ref:`max_task_retries <actor-fault-tolerance>` depending on whether it is a task or actor.
 
-Using the Memory Monitor
-------------------------
+Worker killing policy
+~~~~~~~~~~~~~~~~~~~~~
 
-Retry policy
-~~~~~~~~~~~~
+The memory monitor will ensure liveness by having at least one task or actor execute for each caller. If the last task of a caller is killed then it will fail the workload. Refer to :ref:`how to address memory issues <addressing-memory-issues>` on how to adjust the workload to make it pass when this happens.
 
-When a task or actor is killed by the memory monitor, it will retry using a separate retry counter based off of ``RAY_task_oom_retries`` instead of the typical number of retries specified by :ref:`max_retries <task-fault-tolerance>` for tasks and :ref:`max_restarts <actor-fault-tolerance>` for actors. The number of memory monitor retries is the same for tasks and actors and defaults to ``15``. To override this value, the environment variable ``RAY_task_oom_retries`` should be set when starting Ray as well as the application.
+When a worker needs to be killed, the policy first prioritizes tasks that are retriable, i.e. when ``max_retries`` or ``max_restarts`` is > 0. This is done to minimize workload failure.
 
-Let's create an application oom.py that will trigger the out-of-memory condition.
+.. note::
 
-.. literalinclude:: ../doc_code/ray_oom_prevention.py
-      :language: python
-      :start-after: __oom_start__
-      :end-before: __oom_end__
+    Actors by default are not retriable since :ref:`max_restarts <actor-fault-tolerance>` defaults to 0, therefore tasks are preferred to actors when it comes to what gets killed first. Also, actors will retry based on its ``max_restarts`` settings, even when it is killed by the memory monitor.
```

Comment:
> actors will retry based on its ``max_restarts`` settings, even when it is killed by the memory monitor.

Bit confused about what this means. So tasks OOM retry is not based on max_retries, but actors are based on `max_restarts`?

Reply:
Maybe it is easier to understand if you have some table or bullet points to compare actor/task policy difference